components/FormInput/SharedFormInput.js

```diff
@@ -60,12 +60,13 @@ export default class SharedFormInput extends Component {
       error,
       value,
       label,
+      labelOutside,
       ...remainingProps,
     } = this.props;
     const { focus, id } = this.state;
 
-    const active = value && !!value.length;
-    const isInsideVariant = !!placeholder;
+    const active = labelOutside ? !!placeholder : (value && !!value.length);
+    const isInsideVariant = labelOutside ? false : !!placeholder;
```

Comment:
Not sure, but the logic generally could do with a cleanup so it's more obvious what's happening. 

The `active` className and stuff is only used for the inputs with the labels outside too...

Maybe something more like:
```jsx
const hasPlaceholderAndValue = placeholder && (value && !!value.length);
const labelIsOutside = labelOutside || hasPlaceholderAndValue;

const outerCSS = [
  css.container,
  required
    ? css.required
    : null,
  labelIsOutside
    ? css.labelInside
    : css.labelOutside,
  error
    ? css.error
    : null,
  focus
    ? css.focus
    : null,
  borderless
    ? css.borderless
    : null,
].join(' ');
```